gateway-provider-security-hadoopauth/src/main/java/org/apache/knox/gateway/hadoopauth/filter/HadoopAuthFilter.java

```diff
@@ -201,6 +246,86 @@ public Principal getUserPrincipal() {
     super.doFilter(filterChain, request, response);
   }
 
+  /**
+   * A helper method that checks whether request contains
+   * unauthenticated path
+   * @param request
+   * @return
+   */
+  private boolean doesRequestContainUnauthPath(final ServletRequest request) {
+    for (final String path : unAuthenticatedPaths) {
+      /* make sure the path matches EXACTLY to prevent auth bypass */
+      if (((HttpServletRequest) request).getPathInfo().equals(path)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * A function that let's configured unauthenticated path requests to
+   * pass through without requiring authentication.
+   * An anonymous subject is created and the request is audited.
+   *
+   * Fail gracefully by logging error message.
+   * @param request
+   * @param response
+   * @param chain
+   */
+  private void checkForUnauthenticatedPaths(final ServletRequest request,
```

Comment:
This method does not appear to be "checking" for unauthenticated paths. I think it could be more accurately named something like `handleUnauthenticatedPath` or `bypassAuthForPath`.

Reply:
Nevermind...I saw the follow-on commit.

Reply:
I believe this was for older commit. This is fixed now.